nav2_behavior_tree/plugins/condition/is_battery_low_condition.cpp

```diff
@@ -27,7 +27,25 @@ IsBatteryLowCondition::IsBatteryLowCondition(
   battery_topic_("/battery_status"),
   min_battery_(0.0),
   is_voltage_(false),
-  is_battery_low_(false)
+  is_battery_low_(false),
+  initialized_(false)
+{
+}
+
+BT::NodeStatus IsBatteryLowCondition::tick()
+{
+  if (!initialized_) {
+    initialize();
+  }
+
+  callback_group_executor_.spin_some();
+  if (is_battery_low_) {
+    return BT::NodeStatus::SUCCESS;
+  }
+  return BT::NodeStatus::FAILURE;
+}
+
+void IsBatteryLowCondition::initialize()
```

Comment:
The creation of the callback ground / subscription needs to happen in the the constructor to give it a little extra time for DDS to deal with discovery before the first tick.

Reply:
Ditto in a few different nodes

Reply:
working on it

Reply:
In this case, we decided to stick with the old implementation and here's why:
- we have to create a subscription in the constructor, and this node will be permanently subscribed to only one topic => the user has to use this node as follows:  
`<IsBatteryLowCondition topic_name="some_topic">`, not like this: 
`<IsBatteryLowCondition topic_name="{some_topic}">`

So, I'll revert changes in all nodes where we create a subscription in the constructor